### ISSUE TITLE
Settings --> Checkbox names are not displayed

### ISSUE THREAD

**Issue by kramola-RU** [farblos/copy-on-select-2#15]
Hi !
Win11 22635.3066
Firefox 124.02 --> **dark theme**
Copy on Select 2 - **2.8**

**Checkbox names are not displayed in the extension settings.**

![cos](https://github.com/user-attachments/assets/9dd438e6-701b-492e-8cfc-9b6fdd30aa04)

The font merges with the background when a dark theme is selected.
Could you please fix this Issue ?

**Comment by farblos** [comment]
Uh, that's certainly suboptimal, thanks for reporting.

I've tested with "regular" dark mode (fresh FF profile -> about:preferences -> Website appearance -> Dark), and with that the fonts on the option page are inverted (white-on-black) properly.

Do you have any special settings or theme to get your dark mode?

And what Firefox version are you using?

**Comment by kramola-RU** [comment]
> Do you have any special settings or theme to get your dark mode?

No.

> And what Firefox version are you using?

Firefox x64 v.124.02

-------------------
Drag-Select Link Text -- Settings (for example)
![DSLT](https://github.com/user-attachments/assets/45b51894-d996-486c-bfec-586c27b30b54)

**Comment by farblos** [comment]
Yep, I added some extra CSS to let the c-o-s-2 preferences also turn dark.  (In my opinion it's a FF bug that it does not render the add-on preferences according to the overall theme.  Ungoogled Chromium does that automatically, for example.  It even inverts the add-on icon ...)

If I cannot easily reproduce or fix that extra CSS magic I might simply back it out and leave the option page black-on-white, hoping that Mozilla cares about that eventually.

Just one last question:

>>  Do you have any special settings or theme to get your dark mode?

> No.

What exactly is your FF appearance setting, then?  Just "Dark" or "Automatic" plus the OS having a dark theme?

**Comment by farblos** [comment]
Ok, I have been able to reproduce your issue, actually very easily - just installed your FF version (on Linux x64) and used standard mode "Dark".  On all other FF versions that I have tested (78, 128, nightly) the add-on preferences look OK, only on that version - not so.  Will check what can be done here.

**Comment by farblos** [comment]
So the right thing here according to [this link](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Options_pages#specifying_the_options_page) is to specify a `<meta name="color-scheme" content="dark light" />` meta tag in the option page.  [This bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1888866) also provides some background information.

And that seems to work for FF nightly at least: With that tag, the addon preferences follow the overall browser theme **and** stay readable in light and dark themes.  For FF 124 that is not the case, the addon preferences stay light regardless of browser theme.  But at least they stay readable, so that should be the way to go.

Will provide a fix along those lines, might take some time though due to Christmas holidays ...

**Comment by kramola-RU** [comment]
> What exactly is your FF appearance setting, then? Just "Dark" or "Automatic"

"Automatic"

> plus the OS having a dark theme?

![w](https://github.com/user-attachments/assets/2e435b46-7e7c-4e0f-a907-34c7f8f872b8)

> Will provide a fix.......

Thank you very much!

**Comment by farblos** [comment]
Please try the attached [copy_on_select_2-2.8.2.zip](https://github.com/user-attachments/files/18246912/copy_on_select_2-2.8.2.zip) [~copy_on_select_2-2.8.1.zip~](https://github.com/user-attachments/files/18246412/copy_on_select_2-2.8.1.zip) after renaming it to `copy_on_select_2-2.8.2.xpi`.  It is signed already and should be installable also in all non-nightly Firefoxes.

Thanks!

**EDIT**: Uploaded another version 2.8.2 of the XPI.

**Comment by kramola-RU** [comment]
> Please try the attached [copy_on_select_2-2.8.2.zip](https://github.com/user-attachments/files/18246912/copy_on_select_2-2.8.2.zip)

Excellent !
Thank you very much!

![037](https://github.com/user-attachments/assets/3d5c1f0f-989e-4057-be83-5a3e89b732d4)

**Comment by farblos** [comment]
Thanks for testing.

Дед мороз принёс [версию 2.9](https://git.sr.ht/~jschmidt/copy-on-select-2/refs/v2.9) ... с Новым Годом!

Please re-open if there are still issues.

**Comment by kramola-RU** [comment]
> Дед мороз принёс [версию 2.9](https://git.sr.ht/~jschmidt/copy-on-select-2/refs/v2.9) ... с Новым Годом!

Спасибо! С Новым Годом Вас !



> Please re-open if there are still issues.

Not issue, but a small visual artifact (when zoom <100%):
![95%](https://github.com/user-attachments/assets/4e0fe616-63c5-4499-8fac-3c2c1540fbdc)

When 105% or more --> OK (no white strip)
![105%](https://github.com/user-attachments/assets/7d61b2e4-8ada-4333-b00f-5d46665cb218)

**Comment by farblos** [comment]
> Not issue, but a small visual artifact (when zoom <100%)

Good catch, but I think this is nothing I can do about. The least problem being here that I cannot even reproduce it - I can zoom only in steps of 10% and with that I cannot get that stripe.  I guess in your case some white background from an enclosing component shows through.  If I reload the option page with Ctrl+R in dark mode, I can also see a brief white flash, for example.

So there are/were definetely some layout issues with these embedded option pages in FF, also for example:

- https://bugzilla.mozilla.org/show_bug.cgi?id=1881055#cs-9
- https://bugzilla.mozilla.org/show_bug.cgi?id=1939284

So if you think "your" layout artifact is worth reporting, you might want to check whether this is still present in newer FF versions (I guess not) and, if so, report it on bugzilla.

**Comment by kramola-RU** [comment]
> I think this is nothing I can do about

This is not a critical artifact at all. Let's leave it as it is.
Thank you very much.
